test/python/test_kinetics.py

```diff
@@ -18,179 +20,183 @@
-            self.phase.equilibrium_constants[ix])
-
-    def test_species_rates(self):
-        nu_p = self.phase.product_stoich_coeffs
-        nu_r = self.phase.reactant_stoich_coeffs
-        creation = (np.dot(nu_p, self.phase.forward_rates_of_progress) +
-                    np.dot(nu_r, self.phase.reverse_rates_of_progress))
-        destruction = (np.dot(nu_r, self.phase.forward_rates_of_progress) +
-                       np.dot(nu_p, self.phase.reverse_rates_of_progress))
-
-        self.assertArrayNear(self.phase.creation_rates, creation)
-        self.assertArrayNear(self.phase.destruction_rates, destruction)
-        self.assertArrayNear(self.phase.net_production_rates,
-                             creation - destruction)
-
-    def test_reaction_deltas(self):
-        self.assertArrayNear(self.phase.delta_enthalpy -
-                             self.phase.delta_entropy * self.phase.T,
-                             self.phase.delta_gibbs)
-        self.assertArrayNear(self.phase.delta_standard_enthalpy -
-                             self.phase.delta_standard_entropy * self.phase.T,
-                             self.phase.delta_standard_gibbs)
-
-
-class KineticsFromReactions(utilities.CanteraTest):
+    def test_rates_of_progress(self, phase):
+        assert len(phase.net_rates_of_progress) == phase.n_reactions
+        assert (phase.forward_rates_of_progress -
+                phase.reverse_rates_of_progress) == approx(phase.net_rates_of_progress)
+
+    def test_heat_release(self, phase):
+        hrr = - phase.partial_molar_enthalpies.dot(phase.net_production_rates)
+        assert hrr == approx(phase.heat_release_rate)
+        assert hrr == approx(sum(phase.heat_production_rates))
+
+    def test_rate_constants(self, phase):
+        assert len(phase.forward_rate_constants) == phase.n_reactions
+        ix = phase.reverse_rate_constants != 0.
+        assert (phase.forward_rate_constants[ix] /
+                phase.reverse_rate_constants[ix]) == approx(
+               phase.equilibrium_constants[ix])
+
+    def test_species_rates(self, phase):
+        nu_p = phase.product_stoich_coeffs
+        nu_r = phase.reactant_stoich_coeffs
+        creation = (np.dot(nu_p, phase.forward_rates_of_progress) +
+                    np.dot(nu_r, phase.reverse_rates_of_progress))
+        destruction = (np.dot(nu_r, phase.forward_rates_of_progress) +
+                       np.dot(nu_p, phase.reverse_rates_of_progress))
+
+        assert phase.creation_rates == approx(creation)
+        assert phase.destruction_rates == approx(destruction)
+        assert phase.net_production_rates == approx(creation - destruction)
+
+    def test_reaction_deltas(self, phase):
+        assert (phase.delta_enthalpy -
+                phase.delta_entropy * phase.T) == approx(phase.delta_gibbs)
+
+        assert (phase.delta_standard_enthalpy -
+                phase.delta_standard_entropy * phase.T) == approx(
+               phase.delta_standard_gibbs)
```

Comment:
Borrowing from the "binary operator" example given in [PEP8](https://peps.python.org/pep-0008/#should-a-line-break-before-or-after-a-binary-operator), I'd suggest:
```suggestion
        assert (phase.delta_enthalpy - phase.delta_entropy * phase.T
                == approx(phase.delta_gibbs))

        assert (phase.delta_standard_enthalpy - phase.delta_standard_entropy * phase.T
                == approx(phase.delta_standard_gibbs))
```